assets/prototype/application/utilities/money/test/amountsMatch.test.ts

```diff
@@ -0,0 +1,30 @@
+import { amountsMatch } from '../';
+
+describe('amountsMatch', () => {
+	it('checks for equality for same amounts as string', () => {
+		const amount1 = '5.623';
+		const amount2 = '5.623';
+
+		const result = amountsMatch(amount1, amount2);
+
+		expect(result).toBe(true);
+	});
+
+	it('checks for equality for same amounts as string and number', () => {
+		const amount1 = '7.453';
+		const amount2 = 7.453;
+
+		const result = amountsMatch(amount1, amount2);
+
+		expect(result).toBe(true);
+	});
+
+	it('checks for unequality for different amounts', () => {
+		const amount1 = '8.453';
+		const amount2 = '7.453';
+
+		const result = amountsMatch(amount1, amount2);
+
+		expect(result).toBe(false);
+	});
+});
```

Comment:
I think low-level utilities like this could benefit from having more tests written in order to confirm behaviour under any circumstances and edge cases that could potentially occur at runtime (like a var being undefined).
I also think this is a good place to run tests in a loop in order to simplify things, ex:

```js
const testCases = [
	{ 
		desc: 'checks for equality for same amounts as string',
		amount1: '5.623',
		amount2: '5.623',
		result : true,
	},
	{ 
		desc: 'checks for equality for same amounts as string and number',
		amount1: '7.453',
		amount2: 7.453,
		result : true,
	},
	{ 
		desc: 'checks for unequality for different amounts',
		amount1: '8.453',
		amount2: '7.453',
		result : false,
	},
	{ 
		desc: 'returns false when one value is not a number',
		amount1: 'cabbage',
		amount2: '7.453',
		result : false,
	},
	{ 
		desc: 'returns false when one value is null',
		amount1: null,
		amount2: '0',
		result : false,
	},
	// additional tests cases to cover other possibilities that can occur at runtime
];

describe('amountsMatch', () => {
	for (const testCase of testCases) {
		it( testCase.desc, () => {
			const result = amountsMatch( testCase.amount1,  testCase.amount2);
			expect(result).toBe( testCase.result );
		});
	}
}
```